external_server/server.py

```diff
@@ -0,0 +1,879 @@
+from external_server.models.structures import (
+    GeneralErrorCode,
+    DisconnectTypes,
+    TimeoutType,
+)
+from external_server.models.devices import DevicePy, KnownDevices, device_repr
+from external_server.models.events import (
+    EventType,
+    Event as _Event,
+    EventQueue as _EventQueue,
+)
+from external_server.server_module.server_module import ServerModule as _ServerModule
+from external_server.models.structures import HandledCommand as _HandledCommand
+
+
+eslogger = _ESLogger(LOGGER_NAME)
+carlogger = _CarLogger(LOGGER_NAME)
+
+
+class ServerState(enum.Enum):
+    UNINITIALIZED = enum.auto()
+    CONNECTED = enum.auto()
+    INITIALIZED = enum.auto()
+    RUNNING = enum.auto()
+    STOPPED = enum.auto()
+    ERROR = enum.auto()
+
+
+DeviceStatusName = {
+    _Status.CONNECTING: "CONNECTING",
+    _Status.RUNNING: "RUNNING",
+    _Status.DISCONNECT: "DISCONNECT",
+    _Status.ERROR: "ERROR",
+}
+
+
+StateTransitionTable: dict[ServerState, set[ServerState]] = {
+    ServerState.UNINITIALIZED: {
+        ServerState.ERROR,
+        ServerState.STOPPED,
+        ServerState.CONNECTED,
+    },
+    ServerState.CONNECTED: {
+        ServerState.ERROR,
+        ServerState.STOPPED,
+        ServerState.INITIALIZED,
+    },
+    ServerState.INITIALIZED: {
+        ServerState.ERROR,
+        ServerState.STOPPED,
+        ServerState.RUNNING,
+    },
+    ServerState.RUNNING: {
+        ServerState.ERROR,
+        ServerState.STOPPED,
+    },
+    ServerState.STOPPED: {
+        ServerState.UNINITIALIZED,
+    },
+    ServerState.ERROR: {
+        ServerState.UNINITIALIZED,
+    },
+}
+
+
+ExternalClientMessage = _Connect | _Status | _CommandResponse
+
+
+class ExternalServer:
+
+    def __init__(self, config: ServerConfig) -> None:
+        self._car_servers: dict[str, CarServer] = {}
+        for car_name in config.cars:
+            self._car_servers[car_name] = CarServer(CarConfig.from_server_config(car_name, config))
+        self._company = config.company_name
+
+    @property
+    def company(self) -> str:
+        return self._company
+
+    def car_servers(self) -> dict[str, CarServer]:
+        return self._car_servers.copy()
+
+    def start(self) -> None:
+        threads: dict[str, threading.Thread] = {}
+        for car in self._car_servers:
+            threads[car] = threading.Thread(target=self._car_servers[car].start)
+        for t in threads.values():
+            t.start()
+
+    def stop(self, reason: str = "") -> None:
+        for e in self.car_servers().values():
+            e.stop(reason)
+
+    def tls_set(self, ca_certs: str, certfile: str, keyfile: str) -> None:
+        for car_server in self._car_servers.values():
+            car_server.tls_set(ca_certs, certfile, keyfile)
+
+
+class CarServer:
+
+    def __init__(self, config: CarConfig) -> None:
+        self._running = False
+        self._config = config
+        self._car = config.car_name
```

Comment:
What would be the reason?

Reply:
Consistency. Also it is something which can be part of public API because it identifies the CarServer. 

For example, ExternalServcer class exposes company_name as property.

Reply:
There currently is no reason to access the car name of the car server, outside the logs (where it is done already). 

The company property of ExternalServer class was removed - it was not used anywhere.